apps/minifront/src/state/swap/price-history.ts

```diff
@@ -26,20 +27,18 @@ export const createPriceHistorySlice = (): SliceCreator<PriceHistorySlice> => (s
     const { assetIn, assetOut } = get().swap;
     const startMetadata = getMetadataFromBalancesResponseOptional(assetIn);
     const endMetadata = assetOut;
-    void sendCandlestickDataRequest(
+    void sendCandlestickDataRequests(
       { startMetadata, endMetadata },
-      // there's no UI to set limit yet, and most ranges don't always happen to
-      // include price records. 2500 at least scales well when there is data
-      2500n,
+      // there's no UI to set limit yet, and any given range won't always happen
+      // to include price records.
+      PRICE_RELEVANCE_THRESHOLDS.default * 2n,
       ac.signal,
-    ).then(data => {
-      if (data) {
-        set(({ swap }) => {
-          swap.priceHistory.startMetadata = startMetadata;
-          swap.priceHistory.endMetadata = endMetadata;
-          swap.priceHistory.candles = data;
-        });
-      }
+    ).then(candles => {
+      set(({ swap }) => {
+        swap.priceHistory.startMetadata = startMetadata;
+        swap.priceHistory.endMetadata = endMetadata;
+        swap.priceHistory.candles = candles;
+      });
```

Comment:
I wonder if zQuery could assist in the fetch lifecycle of this slice & the useEffects in the component

Reply:
will make zquery adaption a separate issue/pr